src/main/kotlin/io/libp2p/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -25,9 +27,11 @@ import org.apache.logging.log4j.core.config.Configurator
 import spipe.pb.Spipe
 import java.util.concurrent.CompletableFuture
 
-class NoiseXXSecureChannel(val localKey: PrivKey, val localDHState: DHState, val remoteDHState: DHState, val role: Int) :
+open class NoiseXXSecureChannel(val localKey: PrivKey, val privateKey25519: ByteArray) :
     SecureChannel {
-    private val logger = LogManager.getLogger(NoiseXXSecureChannel::class.java.name + ":" + role)
+    private var role: Int = HandshakeState.RESPONDER
```

Comment:
Can we make it `lateinit` as well, cause referring this property before `initChannel` is invalid situation

Reply:
Good catch. Since lateinit doesn't work with primitive types, I'm changing it to AtomicInteger for the time being.

Reply:
done